template/runtime/src/lib.rs

```diff
@@ -799,20 +798,45 @@ impl_runtime_apis! {
 				None
 			};
 
-			let gas_limit = gas_limit.min(u64::MAX.into());
-			let transaction_data = TransactionData::new(
-				TransactionAction::Call(to),
-				data.clone(),
-				nonce.unwrap_or_default(),
-				gas_limit,
-				None,
-				max_fee_per_gas,
-				max_priority_fee_per_gas,
-				value,
-				Some(<Runtime as pallet_evm::Config>::ChainId::get()),
-				access_list.clone().unwrap_or_default(),
-			);
```

Comment:
Sorry for the delay in replying(just come back from holiday). I come up with another idea is that we can pass default value to the TransactionData::new for these field with Option<T> type, such as max_fee_per_gas and max_priority_fee_per_gas, instead of None. This way can also meet our needs and looks cleaner. What do you think? @librelois